pulsar-broker/src/main/java/com/yahoo/pulsar/broker/service/BrokerService.java

```diff
@@ -418,6 +431,15 @@ public PulsarClient getReplicationClient(String cluster) {
     }
 
     private CompletableFuture<Topic> createPersistentTopic(final String topic) throws RuntimeException {
+        try {
+            return createPersistentTopic(topic, false);
+        } catch (TooManyRequestsException e) {
+            return FutureUtil.failedFuture(e);
+        }
+    }
+
+    private CompletableFuture<Topic> createPersistentTopic(final String topic, boolean failOnTooManyConcurrentTopics)
```

Comment:
Instead of having these 2 methods, I was thinking something like:

```java
CompletableFuture<Topic> createPersistentTopic(final String topic) {
    // Creates future and call method below or queue the request
}


void createPersistentTopic(String topic, CompletableFuture<Topic> topicFuture) {
   // To be called when we are ready to load the topic
} 
```